dandi/cli/cmd_service_scripts.py

```diff
@@ -0,0 +1,103 @@
+from ..exceptions import NotFoundError
+
+
+@click.group()
+def service_scripts() -> None:
+    """Various utility operations"""
+    pass
+
+
+@service_scripts.command()
+@click.option("--diff", is_flag=True, help="Show diffs of old & new metadata")
+@click.option(
+    "--when",
+    type=click.Choice(["newer-schema-version", "always"]),
+    default="newer-schema-version",
+    help="When to re-extract an asset's metadata",
+    show_default=True,
+)
+@click.argument("url")
+@map_to_click_exceptions
+def reextract_metadata(url: str, diff: bool, when: str) -> None:
+    """
+    Recompute & update the metadata for NWB assets on a remote server.
+
+    Running this command requires the fsspec library to be installed with the
+    `http` extra (e.g., `pip install "fsspec[http]"`).
+    """
+    from ..metadata import nwb2asset  # Avoid heavy import at top level
+
+    parsed_url = parse_dandi_url(url)
+    if parsed_url.version_id not in (None, "draft"):
+        raise click.UsageError("URL cannot point to a published Dandiset")
+    parsed_url.version_id = "draft"
+    current_schema_version = Version(DANDI_SCHEMA_VERSION)
+    with parsed_url.navigate(authenticate=True, strict=True) as (_, _, assets):
+        for asset in assets:
+            if PurePosixPath(asset.path).suffix.lower() != ".nwb":
+                lgr.info(
+                    "Asset %s (%s) is not NWB; skipping", asset.identifier, asset.path
+                )
+            assert isinstance(asset, RemoteBlobAsset)
+            lgr.info("Processing asset %s (%s)", asset.identifier, asset.path)
+            if when == "always":
+                do_reextract = True
+            else:
+                try:
+                    sv = asset.get_raw_metadata()["schemaVersion"]
+                except KeyError:
+                    do_reextract = True
+                else:
+                    schemaVersion = Version(sv)
+                    if schemaVersion < current_schema_version:
+                        do_reextract = True
+                    elif schemaVersion == current_schema_version:
+                        lgr.info(
+                            "Asset's schemaVersion is up to date; not reextracting"
+                        )
+                        do_reextract = False
+                    else:
+                        lgr.warning(
+                            "schemaVersion of asset %s (%s) is %r, higher than"
+                            " current schema version %r",
+                            asset.identifier,
+                            asset.path,
+                            sv,
+                            DANDI_SCHEMA_VERSION,
+                        )
+                        do_reextract = False
+            if do_reextract:
+                try:
+                    digest = asset.get_digest()
+                except NotFoundError:
+                    digest = None
+                lgr.info("Extracting new metadata for asset")
```

Comment:
```suggestion
                lgr.info(
                    "Extracting new metadata for asset with schemaVersion %s",
                    schemaVersion
                )
```

Reply:
any objection against making this message more informative?  otherwise user would not know from which version we are "upgrading"

Reply:
For `--when newer-schema-version`, there are already messages emitted giving the metadata's current schemaVersion when deciding whether to re-extract.  For `--when always`, do you really care what the old version was?

Reply:
I might, e.g. for later reanalysis of the logs on how many were updated or completely overridden. Overall I do not see cons from including it here even though indeed might duplicate... but ok, let's forget about it for now since already merged ;)